java/src/jmri/jmrit/operations/trains/TrainsTableModel.java

```diff
@@ -522,7 +522,12 @@ public synchronized void propertyChange(PropertyChangeEvent e) {
                 log.debug("Update train table row: {} name: {}", row, train.getName());
             }
             if (row >= 0) {
-                _table.scrollRectToVisible(_table.getCellRect(row, 0, true));
```

Comment:
Thanks, that seems to fix the lockup.

 

Dan

 

From: Randall Wood [mailto:notifications@github.com] 
Sent: Friday, March 25, 2016 11:52 AM
To: JMRI/JMRI <JMRI@noreply.github.com>
Cc: Daniel Boudreau <daboudreau@hotmail.com>
Subject: Re: [JMRI] Eliminate nasty Java lockup. (#1030)

 

In java/src/jmri/jmrit/operations/trains/TrainsTableModel.java <https://github.com/JMRI/JMRI/pull/1030#discussion_r57456876> :

> @@ -522,7 +522,12 @@ public synchronized void propertyChange(PropertyChangeEvent e) {
>                  log.debug("Update train table row: {} name: {}", row, train.getName());
>              }
>              if (row >= 0) {
> -                _table.scrollRectToVisible(_table.getCellRect(row, 0, true));

Wrap this in SwingUtilities.invokeLater() JavaDoc <http://docs.oracle.com/javase/8/docs/api/javax/swing/SwingUtilities.html#invokeLater-java.lang.Runnable-> 

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub <https://github.com/JMRI/JMRI/pull/1030/files/899f855d0f2b2d499b4a7edb17b03727804f8ad4#r57456876>   <https://github.com/notifications/beacon/ANw2pwNtS0KKV7zg-J9YqB3hjgroBLIfks5pxASvgaJpZM4H4y1k.gif>